aggregator/placement_mgr.go

```diff
@@ -214,14 +214,14 @@ func (mgr *placementManager) placementWithLock() (placement.Placement, error) {
 		mgr.metrics.activeStagedPlacementErrors.Inc(1)
 		return nil, err
 	}
-	defer onStagedPlacementDoneFn()
```

Comment:
nit: maybe add a comment on why we don't use `defer` here?

Reply:
Sounds good, will do